src/bootstrap/dist.rs

```diff
@@ -482,12 +482,12 @@ impl Step for Rustc {
             builder.install(&builder.rustdoc(compiler), &image.join("bin"), 0o755);
 
             // Copy runtime DLLs needed by the compiler
-            if libdir != "bin" {
-                for entry in builder.read_dir(&src.join(libdir)) {
+            if libdir.file_name().map_or(true, |f| f != "bin") {
```

Comment:
Why would there not be a file_name on the libdir path? I would expect this to unwrap, I think.

Reply:
Fixed.